docs/tutorials/replicated-stateful-application/mysql-statefulset.yaml

```diff
@@ -0,0 +1,144 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: mysql
+spec:
+  serviceName: mysql
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: mysql
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
+          {
+            "name": "init-mysql",
+            "image": "mysql:5.7",
+            "command": ["bash", "-c", "
+              set -ex\n
+              # mysqld --initialize expects an empty data dir.\n
+              rm -rf /mnt/data/lost+found\n
+              # Copy conf.d from config-map to emptyDir.\n
+              cp /mnt/config-map/* /mnt/conf.d/\n
+              # Generate mysql server-id from pod ordinal index.\n
+              [[ `hostname` =~ -([0-9]+)$ ]] || exit 1\n
+              echo [mysqld] > /mnt/conf.d/server-id.cnf\n
+              echo server-id=$((100 + ${BASH_REMATCH[1]})) >> /mnt/conf.d/server-id.cnf\n
```

Comment:
MySQL server id 0 is reserved, so we need some offset on the ordinal index. I didn't want to use something like `ordinal+1` because it would be easy to see server id 1 and mistakenly assume that corresponds to ordinal 1. I'll add a comment somewhere nearby to make this less mysterious.